tests/span_tests.cpp

```diff
@@ -483,6 +483,66 @@ SUITE(span_tests)
 #endif
     }
 
+    TEST(from_unique_pointer_construction)
+    {
+        {
+            auto arr = new int[4];
+            auto ptr = std::unique_ptr<int>(arr);
+
+            for (auto i = 0; i < 4; i++)
+                arr[i] = i + 1;
+
+            {
+                span<int> s{arr, 4};
+                CHECK(s.length() == 4 && s.data() == ptr.get());
+                CHECK(s[0] == 1 && s[1] == 2);
+            }
+
+            {
+                span<int> s{std::unique_ptr<int>{nullptr}, 0};
+                CHECK(s.length() == 0 && s.data() == nullptr);
+            }
+        }
+
+        {
+            auto arr = std::make_unique<int[]>(4);
+
+            for (auto i = 0; i < 4; i++)
+                arr[i] = i + 1;
+
+            {
+                span<int> s{arr, 4};
+                CHECK(s.length() == 4 && s.data() == arr.get());
+                CHECK(s[0] == 1 && s[1] == 2);
+            }
+
+            {
+                span<int> s{std::unique_ptr<int>{nullptr}, 0};
+                CHECK(s.length() == 0 && s.data() == nullptr);
+            }
+        }
+    }
+
+    TEST(from_shared_pointer_construction)
+    {
+        auto arr = new int[4];
+        auto ptr = std::shared_ptr<int>(arr);
```

Comment:
I'm pretty sure this will use the wrong `delete` unless you provide a custom deleter argument.